### PR TITLE
add role to create routes

### PR DESCRIPTION
**Pull request by xcoulon**
this is going to be used by "perf" tests to access
the `/metrics` endpoint

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Review by MatousJobanek — APPROVED** [review]
I guess that you need this because you want to create a route for the `/metric` endpoint so you can then access it from the host cluster, right?
I'm curious - wouldn't it be easier to do that as part of the test setup?

**Comment by alexeykazakov** [comment]
Yeah.. If this is needed only for the tests then ideally it should be in the tests only. Unless it's too tricky.

**Comment by MatousJobanek** [comment]
I actually totally forgot that we are already doing it for host-operator, so the code is already there (in e2e repo) and working. Xavier just needs to extend it also for member operator. Let say that it's fine for now and optimize the permissions later if we need it. 
@xcoulon you could just add the needed permissions (`create`) only to namespaced role and not cluster role - that's the biggest difference compared to host operator

**Comment by xcoulon** [comment]
> I actually totally forgot that we are already doing it for host-operator, so the code is already there (in e2e repo) and working. Xavier just needs to extend it also for member operator. Let say that it's fine for now and optimize the permissions later if we need it.
> @xcoulon you could just add the needed permissions (`create`) only to namespaced role and not cluster role - that's the biggest difference compared to host operator

done in 9a1c2c8

File: pkg/controller/idler/idler_controller.go [hunk 1]
```diff
@@ -340,8 +340,7 @@ func nextPodToBeKilledAfter(idler *toolchainv1alpha1.Idler) *time.Duration {
 	}
 	var d time.Duration
 	for _, pod := range idler.Status.Pods {
-		whenToKill := pod.StartTime.Add(time.Duration(idler.Spec.TimeoutSeconds+1) * time.Second)
-		killAfter := whenToKill.Sub(time.Now())
+		killAfter := time.Until(pod.StartTime.Add(time.Duration(idler.Spec.TimeoutSeconds+1) * time.Second))
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply pkg/controller/idler/idler_controller.go]: yes, this change is now in https://github.com/codeready-toolchain/member-operator/pull/209/files#diff-0ddf448ea61959821185b66811f19adbd7a7cc10c772d44722bc0cffa81a425bR343 instead

[the remark this answers is not in the document]
Comment (xcoulon) [reply pkg/controller/idler/idler_controller.go]: reverted in 70d1e4c

**Review by tinakurian — APPROVED** [review]

**Review by MatousJobanek — APPROVED** [review]
Thanks a lot for moving it to `role.yaml` file

**Pull request merged**